### ISSUE TITLE
Can't start gnome session when selecting "switch to desktop" on steam UI

### ISSUE THREAD

**Issue by imadnyc** [Jovian-Experiments/Jovian-NixOS#202]
Plasma works fine though. 

This discussion https://github.com/Jovian-Experiments/Jovian-NixOS/discussions/179#discussioncomment-7362351 is where I explained it in more detail. If I instead use `xserver.desktopManager.plasma5.enable = true` and set `jovian.steam.desktopSession = "plasma"`, the plasma session works fine. This seems to be an issue with only me. 

@DylanRJohnston-FZ says they have a config that worked fine, but copying his config and removing imports that I didn't have results in the same issue. This is against the latest stable nixpkgs. 

Strangely, in their config, their `sessionName = "gnome-wayland"`, which is not a valid session name according to `config.services.xserver.displayManager.sessionData.sessionNames`, yet it still worked. 

I'm unsure what to attach that would give more information.

**Comment by imadnyc** [comment]
Here are the logs from `journalctl -u greetd`

```
Nov 06 20:03:02 nix-deck systemd[1]: Started greetd.service.
Nov 06 20:03:08 nix-deck greetd[1525]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:03:08 nix-deck greetd[1525]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:03:08 nix-deck greetd[1525]: pam_unix(greetd:session): session closed for user jovian-greeter
Nov 06 20:03:08 nix-deck greetd[1618]: pam_unix(greetd:session): session opened for user dre(uid=1000) by (uid=0)
Nov 06 20:03:08 nix-deck greetd[1618]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:03:59 nix-deck greetd[3687]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:03:59 nix-deck systemd[1]: Stopping greetd.service...
Nov 06 20:03:59 nix-deck systemd[1]: greetd.service: Deactivated successfully.
Nov 06 20:03:59 nix-deck systemd[1]: Stopped greetd.service.
-- Boot c175c009dd67485199c6c55c2747314b --
Nov 06 20:05:52 nix-deck systemd[1]: Started greetd.service.
Nov 06 20:05:57 nix-deck greetd[1521]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:05:57 nix-deck greetd[1521]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:05:58 nix-deck greetd[1521]: pam_unix(greetd:session): session closed for user jovian-greeter
Nov 06 20:05:58 nix-deck greetd[1556]: pam_unix(greetd:session): session opened for user dre(uid=1000) by (uid=0)
Nov 06 20:05:58 nix-deck greetd[1556]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:08:00 nix-deck greetd[3720]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:08:00 nix-deck greetd[3720]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:08:00 nix-deck greetd[3720]: pam_unix(greetd:session): session closed for user jovian-greeter
Nov 06 20:08:00 nix-deck greetd[3754]: pam_unix(greetd:session): session opened for user dre(uid=1000) by (uid=0)
Nov 06 20:08:00 nix-deck greetd[3754]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:08:01 nix-deck greetd[3754]: pam_unix(greetd:session): session closed for user dre
Nov 06 20:08:02 nix-deck greetd[3852]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:08:02 nix-deck greetd[3852]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:08:02 nix-deck greetd[3852]: pam_unix(greetd:session): session closed for user jovian-greeter
Nov 06 20:08:02 nix-deck greetd[3863]: pam_unix(greetd:session): session opened for user dre(uid=1000) by (uid=0)
Nov 06 20:08:02 nix-deck greetd[3863]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:16:31 nix-deck greetd[6833]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:16:31 nix-deck systemd[1]: Stopping greetd.service...
Nov 06 20:16:31 nix-deck systemd[1]: greetd.service: Deactivated successfully.
Nov 06 20:16:31 nix-deck systemd[1]: Stopped greetd.service.
-- Boot 01a525e19dbc480a8e2976c189d0d489 --
Nov 06 20:18:28 nix-deck systemd[1]: Started greetd.service.
Nov 06 20:18:33 nix-deck greetd[1552]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:18:34 nix-deck greetd[1552]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:18:34 nix-deck greetd[1552]: pam_unix(greetd:session): session closed for user jovian-greeter
Nov 06 20:18:34 nix-deck greetd[1586]: pam_unix(greetd:session): session opened for user dre(uid=1000) by (uid=0)
Nov 06 20:18:34 nix-deck greetd[1586]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:22:23 nix-deck greetd[3655]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:22:23 nix-deck greetd[3655]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:22:23 nix-deck greetd[3655]: pam_unix(greetd:session): session closed for user jovian-greeter
Nov 06 20:22:23 nix-deck greetd[3689]: pam_unix(greetd:session): session opened for user dre(uid=1000) by (uid=0)
Nov 06 20:22:23 nix-deck greetd[3689]: gkr-pam: couldn't unlock the login keyring.
Nov 06 20:22:24 nix-deck greetd[3689]: pam_unix(greetd:session): session closed for user dre
Nov 06 20:22:25 nix-deck greetd[3782]: pam_unix(greetd:session): session opened for user jovian-greeter(uid=989) by (uid=0)
Nov 06 20:22:25 nix-deck greetd[3782]: gkr-pam: gnome-keyring-daemon started properly
Nov 06 20:22:25 nix-deck greetd[3782]: pam_unix(greetd:session): session closed for user jovian-greeter
Nov 06 20:22:25 nix-deck greetd[3793]: pam_unix(greetd:session): session opened for user dre(uid=1000) by (uid=0)
Nov 06 20:22:25 nix-deck greetd[3793]: gkr-pam: gnome-keyring-daemon started properly
```

When I try to open it, it says that it can't unlock the login keyring `Nov 06 20:22:23 nix-deck greetd[3689]: gkr-pam: couldn't unlock the login keyring.`

**Comment by imadnyc** [comment]
Solved by building against unstable, not 23.05.
